src/Sulu/Bundle/SecurityBundle/AccessControl/AccessControlQueryEnhancer.php

```diff
@@ -40,9 +40,13 @@ public function enhance(
         QueryBuilder $queryBuilder,
         UserInterface $user = null,
         int $permission,
-        string $entityClass,
-        string $entityAlias
+        ?string $entityClass,
+        string $entityAlias,
+        ?string $entityClassField = null,
+        ?string $entityIdField = 'id'
```

Comment:
its a bit confusing what parameter does what here. but dont have a better idea, so its okay for me 😇

Reply:
one thing i can think of would be leaving this method like it is and add a second `enhanceWithFields` method:
```
    public function enhance(
        QueryBuilder $queryBuilder,
        UserInterface $user = null,
        int $permission,
        string $entityClassField = null,
        string $entityIdField = 'id'
        string $entityAlias
    )
```